syntax = "proto3";

package foreverbull.backtest;

option go_package = "./pb_gen";

import "google/protobuf/timestamp.proto";

message Period {
    google.protobuf.Timestamp timestamp = 1;
    double PNL = 2;
    double returns = 3;
    double portfolio_value = 4;
    int32 longs_count = 5;
    int32 shorts_count = 6;
    double long_value = 7;
    double short_value = 8;
    double starting_exposure = 9;
    double ending_exposure = 10;
    double long_exposure = 11;
    double short_exposure = 12;
    double capital_used = 13;
    double gross_leverage = 14;
    double net_leverage = 15;
    double starting_value = 16;
    double ending_value = 17;
    double starting_cash = 18;
    double ending_cash = 19;
    double max_drawdown = 20;
    double max_leverage = 21;
    double excess_returns = 22;
    double treasury_period_return = 23;
    double algorithm_period_return = 24;
    optional double algo_volatility = 25;
    optional double sharpe = 26;
    optional double sortino = 27;
    optional double benchmark_period_returns = 28;
    optional double benchmark_volatility = 29;
    optional double alpha = 30;
    optional double beta = 31;
}
